### PR TITLE
Fix 2d corner flow

### PR DESCRIPTION
**Pull request by adigitoleo**
Remove 3/4 cornerflow tests which are pretty redundant, and add some code for pole figures. Add some asserts for the remaining cornerflow test. A few other fixes here and there as well, commits can probably be squashed because it got a bit messy.

### CODE REVIEW

File: src/pydrex/stats.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+from scipy.spatial.transform import Rotation
+
+_RNG = np.random.default_rng()
+
+
+def isotropic_orientations(n_grains, symmetry=(2, 4), rng=_RNG):
+    """Get orientation matrices for an isotropic polycrystal of minerals.
+
+    The optional argument `symmetry` accepts a tuple of integers (a, b)
+    that specify the crystal symmetry system of the mineral.
```
Comment (Patol75) [inline src/pydrex/stats.py:12]: "specifies"

File: src/pydrex/stats.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+def misorientations_random(low, high, symmetry=(2, 4)):
+    """Get expected count of misorientation angles for an isotropic aggregate.
+
+    Estimate the expected number of misorientation angles between grains
+    that would fall within (`low`, `high`) in degrees for an aggregate
+    with randomly oriented grains, where `low` ∈ [0, `high`),
+    and `high` is bounded by the maximum theoretical misorientation angle θmax.
+
+    The optional argument `symmetry` accepts a tuple of integers (a, b)
+    that specify the crystal symmetry system:
```
Comment (Patol75) [inline src/pydrex/stats.py:55]: "specifies"

File: src/pydrex/stats.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        [misorientations_random(θ - 1, θ, symmetry) for θ in range(1, max_θ + 1)]
+    )
+    weights /= weights.sum()
+    # Allocate n_grains worth of identity rotations.
+    orientations = Rotation.from_quat(np.tile([0, 0, 0, 1], (n_grains, 1)))
+    # Set random first orientation.
+    orientations[0] = Rotation.random()
+    axis_map = {0: "X", 1: "Y", 2: "Z"}
+    for n in range(1, n_grains):
+        # The location of each new grain is determined by the misorienatation angle from
```
Comment (Patol75) [inline src/pydrex/stats.py:34]: "misorientation"

File: src/pydrex/visualisation.py [hunk 4]
```diff
@@ -274,3 +281,192 @@ def corner_flow_2d(
...
+    ax.plot(np.cos(_circle_points), np.sin(_circle_points), linewidth=1, color="k")
+    ax.axhline(0, color="k", alpha=0.5)
+    ax.text(1.05, 0.5, ref_axes[0], verticalalignment="center", transform=ax.transAxes)
+    ax.axvline(0, color="k", alpha=0.5)
+    ax.text(
+        0.5, 1.05, ref_axes[1], horizontalalignment="center", transform=ax.transAxes
+    )
+
+
+def poles(orientations, ref_axes="xz", hkl=[1, 0, 0]):
```
Comment (Patol75) [inline src/pydrex/visualisation.py:303]: Just wondering is that is something `pyproj` can help you with?

**Review by Patol75 — CHANGES_REQUESTED** [review]
Looking good, thanks for the nice clean-up! Extremely minor comments and suggestions for you to look at.

File: src/pydrex/visualisation.py (hunk 4 above)
Comment (adigitoleo) [reply src/pydrex/visualisation.py:303]: I looked into it but that requires an unnecessary conversion to spherical coords. Added some docs and comments though in the incoming commit.

**Review by Patol75 — APPROVED** [review]

**Pull request merged**